torch/distributed/rpc/backend_registry.py

```diff
@@ -267,6 +273,56 @@ def _tensorpipe_check_local_device_maps(name, device_count, options):
             f"device_maps = {options.device_maps},\n"
             f"devices = {options.devices}"
         )
+    return local_devices
+
+def _update_group_membership(worker_info, my_devices, reverse_device_map):
+    agent = api._get_current_rpc_agent()
+    ret = agent._update_group_membership(worker_info, my_devices, reverse_device_map, True)
+    return "hi"
+
+def _get_device_infos():
+    if torch.cuda.is_available():
+        device_count = torch.cuda.device_count()
+    else:
+        device_count = 0
+    agent = api._get_current_rpc_agent()
+    opts = agent._get_backend_options()
+    if torch.cuda.is_available():
+        device_count = torch.cuda.device_count()
+    else:
+        device_count = 0
+    return device_count, opts.device_maps, opts.devices
+
+def _set_devices_and_reverse_device_map(agent, my_rank, my_device_maps):
+    print(f"{my_rank}: _set_devices_and_reverse_device_map, {my_device_maps}")
+    my_worker_info = agent.get_worker_info()
+    my_name = my_worker_info.name
+    all_worker_infos = agent.get_worker_infos()
```

Comment:
Just to double check, this is retrieved from store, right?

Reply:
It retrieved from agent's local cache, but ultimately retrieved from store. On TPagent initialization, it retrieves all current workers from store and saves it in a map. `agent.get_worker_infos()` just returns that map. Since this method is called directly after initialization and the group membership is locked at this time, the state of the workers is valid